R/scale-.r

```diff
@@ -107,13 +107,17 @@ Scale <- ggproto("Scale", NULL,
     stop("Not implemented", call. = FALSE)
   },
 
+  #  if scale contains a function, apply it to the default (inverted) scale range
   #  if scale contains a NULL, use the default scale range
   #  if scale contains a NA, use the default range for that axis, otherwise
   #  use the user defined limit for that axis
   get_limits = function(self) {
     if (self$is_empty()) return(c(0, 1))
 
-    if (!is.null(self$limits)) {
+    if (is.function(self$limits)) {
```

Comment:
Since we're adding another condition here, can you please reframe the if statement along these lines:

```R
if (is.null()) {

} else if (is.function()) {

} else if (is.numeric()) {
  
} else {
  stop("Informative error message", call. = FALSE) 
}
```

Reply:
I did rework this conditional to be more intuitive but since `get_limits()` is called by both discrete and continuous scales and can be passed character, numeric, POSIXt and now function classes, I ultimately did not add an explicit `is.numeric` conditional nor an error message. Happy to update further if necessary.